flambe/nlp/language_modeling/sampler.py

```diff
@@ -0,0 +1,129 @@
+from typing import Optional, Sequence, Tuple, Iterator
+import math
+
+import torch
+from torch.utils.data import DataLoader
+
+from flambe.sampler.sampler import Sampler
+
+
+class CorpusSampler(Sampler):
```

Comment:
Maybe it's worth inheriting from `BaseSampler`? I see at least in the `__init__` a lot of shared parameters.

Reply:
Meh barely saves anything. Will go over this during refactor.